### PR TITLE
feat: redesign slash menu

### PR DESCRIPTION
**Pull request by nshelia**
- Add new icons
- Add columns of items
- Add keyboard navigation support (arrow left/right)

Demo: https://www.loom.com/share/4d9d3902dd5c4a8daaf4c3fda85f8d70

### CODE REVIEW

File: src/components/BlockMenu.tsx [hunk 1]
```diff
@@ -115,14 +117,60 @@ class BlockMenu extends React.Component<Props, State> {
...
+
+      if (this.filtered.length) {
+        const columnIndex =
+          Math.floor(this.state.selectedIndex / ITEMS_PER_COLUMN) - 1;
+        const rowIndex = this.state.selectedIndex % 6;
+        let newIndex = columnIndex * ITEMS_PER_COLUMN + rowIndex;
+        if (newIndex < 0) {
+          newIndex = this.state.selectedIndex;
+        }
+        const next = this.filtered[newIndex];
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:170]: I see this code is similar in all cases: ArrowUp, Down, Left, and Right, let's move the logic to a function that can calculate the selected Item for all cases. Make sure the function is outside the component.

File: src/components/BlockMenu.tsx [hunk 2]
```diff
@@ -158,7 +205,7 @@ class BlockMenu extends React.Component<Props, State> {
     }
   };
 
-  insertItem = item => {
+  insertItem = (item) => {
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:208]: A lot of methods in this file are not well typed. Please type the arguments for methods/functions correctly

File: src/components/BlockMenu.tsx [hunk 3]
```diff
@@ -486,37 +535,51 @@ class BlockMenu extends React.Component<Props, State> {
...
+                          <ListItem key={item.title}>
+                            <BlockMenuItem
+                              onClick={() => this.insertItem(item)}
+                              selected={selected}
+                              icon={item.icon}
+                              title={item.title}
+                              shortcut={item.shortcut}
+                            ></BlockMenuItem>
+                          </ListItem>
+                        );
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:561]: Let's move this as a method that will render this

File: src/components/BlockMenu.tsx [hunk 4]
```diff
@@ -486,37 +535,51 @@ class BlockMenu extends React.Component<Props, State> {
...
-                      shortcut={item.shortcut}
-                    ></BlockMenuItem>
-                  </ListItem>
-                );
-              })}
               {items.length === 0 && (
                 <ListItem>
                   <Empty>{dictionary.noResults}</Empty>
                 </ListItem>
               )}
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:572]: image

The box has extra padding or margin at the top part. Let's make it symmetrical.

File: src/components/BlockMenu.tsx [hunk 5]
```diff
@@ -486,37 +535,51 @@ class BlockMenu extends React.Component<Props, State> {
...
+              {shortcut && (
+                <ShortcutWrapper>
+                  {"Shortcut:"} {shortcut}
+                  {markdown && (
+                    <MarkdownText>
+                      {"Markdown:"} {markdown}
+                    </MarkdownText>
+                  )}
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:580]: Let's make these to labels separated and wrapped into containers, for example: `<div>Shorcut: Bla</div><div>Markdown: Bla</div>`. Also, should we move those labels into our dictionary?

**Review by damianrodrigues — CHANGES_REQUESTED** [review]

**Review by damianrodrigues — CHANGES_REQUESTED** [review]
Also in all the files where we've done many changes make sure that we do not have unused variables and types are covered 100%

File: src/components/BlockMenu.tsx [hunk 6]
```diff
@@ -115,14 +117,60 @@ class BlockMenu extends React.Component<Props, State> {
       event.stopPropagation();
 
       if (this.filtered.length) {
-        const prevIndex = this.state.selectedIndex - 1;
+        let prevIndex = this.state.selectedIndex - 1;
+        if (prevIndex < 0) {
+          prevIndex = this.state.selectedIndex;
+        }
         const prev = this.filtered[prevIndex];
```
[the remark this answers is not in the document]
Comment (nshelia) [reply src/components/BlockMenu.tsx:124]: Refactored so the name fits. Note: Arrow Up gets prev item always

File: src/components/BlockMenu.tsx [hunk 7]
```diff
@@ -115,14 +117,60 @@ class BlockMenu extends React.Component<Props, State> {
...
 
         this.setState({
-          selectedIndex: Math.max(
-            0,
-            prev && prev.name === "separator" ? prevIndex - 1 : prevIndex
-          ),
+          selectedIndex: Math.max(0, prev && prevIndex),
```
[the remark this answers is not in the document]
Comment (nshelia) [reply src/components/BlockMenu.tsx:127]: Change logic to do setState when next/prev value is present

**Review by damianrodrigues — APPROVED** [review]
LGTM!

**Pull request merged**